boa/core/metadata.py

```diff
@@ -149,7 +149,7 @@ def get_value(self, in_key: str, default: Any = None, autotype=True) -> Any:
             if autotype and default is None and FIELDS.get(section, {}).get(key):
                 default = FIELDS[section][key]()
 
-        section = self.output.sections.get(section, {})
+        section = self.meta.get(section, {})
```

Comment:
This caused gh-388 since `.get_value` is called from `conda-build` code which expects return types from original `conda_build.metadata.MetaData` (e.g., `str`, not `CondaBuildSpec`).
From a quick glance, I couldn't make out any calls from within `boa` that expect `boa`-specific types here; would be good to have another pair of eyes confirming this..